src/WooCommerce.php

```diff
@@ -249,6 +249,19 @@ public static function cron_remove_back_in_stock() {
     }
   }
 
+  /**
+   * Cron event callback to remove outdated back-in-stock product metadata.
+   */
+  public static function cron_orphans_variants_cleanup() {
+    global $wpdb;
+    $wpdb->query(
+      "DELETE products
+    FROM {$wpdb->posts} products
+    LEFT JOIN {$wpdb->posts} wp ON wp.ID = products.post_parent
+    WHERE wp.ID IS NULL AND products.post_type = 'product_variation';"
+    );
```

Comment:
We can't delete the data directly. We actually need to invoke the WordPress API to delete the posts (or WooCommerce product API if there is a wc_delete_product()), so that third-party services like Elasticsearch are updated accordingly.

Realizing this now, this also applies to the core-standards post meta cleanup cron, which should use delete_post_meta(). But maybe it's not that critical in that one, because post meta cannot appear on its own in third-party services.